cql3/expr/expression.cc

```diff
@@ -2300,5 +2300,28 @@ type_of(const expression& e) {
     }, e);
 }
 
+bool is_single_column_restriction(const expression& e) {
+    if (find_in_expression<unresolved_identifier>(e, [](const auto&) {return true;})) {
+        on_internal_error(expr_logger, format("is_single_column_restriction - expression is not prepared: {}", e));
+    }
+
+    const column_value* the_column = nullptr;
+    bool result = false;
+
+    find_in_expression<column_value>(e,
+        [&](const column_value& cval) -> bool {
+            if (the_column == nullptr) {
+                the_column = &cval;
+                result = true;
+            } else if (the_column->col != cval.col) {
+                result = false;
```

Comment:
It looks incorrect, unless there are some assumptions/invariants I'm not aware of, e.g. that there's  a certain order of expressions. This code looks like it would set result to true for `['a < 7', 'b > 5', 'a < 4']`, because it would only temporarily set `result = false` when visiting b, but go back to `result = true` after `a < 4`. Perhaps you meant to break early if you find another column, or at least use `result &= true` and ` result &= false` when updating `result`?

Reply:
I think it's correct.
`true` is set only in the case where `the_column` is nullptr, so it can happen at most once.
Any future sightings of `column_value` can only set result to false.

I changed this code to be a bit more readable.